src/bootstrap/check.rs

```diff
@@ -115,7 +107,37 @@ pub fn cargo(build: &Build, stage: u32, host: &str) {
     // available.
     cargo.env("CFG_DISABLE_CROSS_TESTS", "1");
 
-    build.run(cargo.env("PATH", newpath));
+    build.run(cargo.env("PATH", &path_for_cargo(build, compiler)));
+}
+
+/// Runs `cargo test` for the rls.
+pub fn rls(build: &Build, stage: u32, host: &str) {
+    let compiler = &Compiler::new(stage, host);
+
+    let mut cargo = build.cargo(compiler, Mode::Tool, host, "test");
+    cargo.arg("--manifest-path").arg(build.src.join("src/tools/rls/Cargo.toml"));
+
+    // Don't build tests dynamically, just a pain to work with
+    cargo.env("RUSTC_NO_PREFER_DYNAMIC", "1");
+
+    let mut dylib_path = dylib_path();
+    dylib_path.insert(0, build.sysroot_libdir(&compiler, host));
+    cargo.env(dylib_path_var(), env::join_paths(&dylib_path).unwrap());
```

Comment:
Oh we've actually got a [helper for this](https://github.com/rust-lang/rust/blob/81734e0e06d24cf580dd5352c64d10110e4d3b7b/src/bootstrap/lib.rs#L733) I think